src/bootstrap/compile.rs

```diff
@@ -532,13 +595,18 @@ impl Step for Rustc {
     const DEFAULT: bool = false;
 
     fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
-        run.never()
+        // NOTE: doesn't use rustc-main to avoid conflicting with `compiler/rustc` in `Assemble`.
+        // If you add a new *direct* dependency to rustc-main, please also update this function.
```

Comment:
It seems like it should be possible for us to do this here by calling in_tree_crates ourselves or adding a helper (krate_deps(...)).